src/main/java/uk/gov/companieshouse/api/accounts/service/impl/CompanyAccountServiceImpl.java

```diff
@@ -85,24 +88,33 @@
         return new ResponseObject<>(ResponseStatus.CREATED, companyAccount);
     }
 
+    @Override
+    public void addLink(String id, LinkType linkType, String link) {
+        CompanyAccountEntity companyAccountEntity = companyAccountRepository.findById(id)
+                .orElseThrow(() -> new MongoException(
+                        "Failed to add link to Company account entity"));
+        CompanyAccountDataEntity companyAccountDataEntity = companyAccountEntity.getData();
+        Map<String, String> map = companyAccountDataEntity.getLinks();
+        map.put(linkType.getLink(), link);
+        companyAccountDataEntity.setLinks(map);
+        companyAccountRepository.save(companyAccountEntity);
```

Comment:
this should be wrapped in a try-catch to handle MongoException and wrap in a DataException to throw

Reply:
as discussed, happy for this class to be reworked in a subsequent PR to fit in with the AbstractService interface